napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,601 @@
+    # (0, 1, 2) and end with (-3, -2, -1).
+    # In the following loop, i represents the center point of the sliding
+    # window. Therefore, and keeping in mind that the stop value of a range in
+    # Python is exclusive, if closed we want i in the range [0, n), while if
+    # open we want i in the range [1, n-1). This can be accomplished by:
+    start = 1 - closed
+    stop = n - 1 + closed
+    for i in range(start, stop):
+        cos_angle = _dot(direction_vectors[i - 1], direction_vectors[i])
+        if cos_angle < cos_miter_limit:
+            bevel_count += 1
+
+    return point_count + bevel_count
+
+
+@njit(cache=True, inline='always')
+def _set_centers_and_offsets(
+    centers: np.ndarray,
+    offsets: np.ndarray,
+    triangles: np.ndarray,
+    vertex: np.ndarray,
+    vec1: np.ndarray,
+    vec2: np.ndarray,
+    vec1_len: float,
+    vec2_len: float,
+    j: int,
+    cos_limit: float,
+    always_bevel: bool,
+) -> int:
+    """Set the centers, offsets, and triangles for a given path and position.
+
+    This function computes the positions of the vertices of the edge triangles
+    at the given path vertex and towards the next path vertex, including, if
+    needed, the miter join triangle overlapping the path vertex.
+
+    If a miter join is needed, this function will add three triangles.
+    Otherwise, it will add two triangles (dividing parallelogram of the
+    next edge into two triangles).
+
+    The positions of the triangle vertices are given as the path vertex
+    (repeated once for each triangle vertex), and offset vectors from that
+    vertex.
+
+    The added triangles "optimistically" index into vertices past the vertices
+    added in this iteration (indices j+3 and j+4).
+
+    Parameters
+    ----------
+    centers : np.ndarray of float
+        Mx2 output array of central coordinates of path triangles.
+    offsets : np.ndarray of float
+        Mx2 output array of the offsets from the central coordinates. Offsets
+        need to be scaled by the line width and then added to the centers to
+        generate the actual vertices of the triangulation.
+    triangles : np.ndarray of int
+        (M-2)x3 output array of the indices of the vertices that will form the
+        triangles of the triangulation.
+    vertex : np.ndarray
+        The path vertex for which the centers, offsets and triangles are
+        being calculated.
+    vec1 : np.ndarray
+        The norm-1 direction vector from the previous path vertex to the
+        current path vertex.
+    vec2 : np.ndarray
+        The norm-1 direction vector from the current path vertex to the next
+        path vertex.
+    vec1_len : float
+        The length of the segment between the previous path vertex and the
+        current path vertex (used for bevel join calculation).
+    vec2_len : float
+        The length of the segment between the current path vertex and the
+        next path vertex (used for bevel join calculation).
```

Comment:
So here we pass half of the vector name when name and description expects full vector length.